Makefile

```diff
@@ -66,13 +71,21 @@ dofmt:
 lint:
 	$(GOPATH)/bin/golangci-lint run --timeout 600s
 
+# Repository does not contain vendor/modules.txt file so re-build with go mod vendor
 .PHONY: build
 build:
-	CGO_ENABLED=0 GOOS=$(shell go env GOOS) GOARCH=$(shell go env GOARCH) go build -mod=vendor -a -ldflags '-X main.vendorVersion='"${DRIVER_NAME}-${GIT_COMMIT_SHA}"' -extldflags "-static"' -o ${GOPATH}/bin/${EXE_DRIVER_NAME} ./cmd/
-
+	CGO_ENABLED=0 GOOS=$(shell go env GOOS) GOARCH=$(shell go env GOARCH) \
+		go mod vendor
+	CGO_ENABLED=0 GOOS=$(shell go env GOOS) GOARCH=$(shell go env GOARCH) \
+		go build -mod=vendor -a \
+		-ldflags '-X main.vendorVersion='"${DRIVER_NAME}-${GIT_COMMIT_SHA}"' -extldflags "-static"' \
+		-o ${GOPATH}/bin/${EXE_DRIVER_NAME} \
+		./cmd/
```

Comment:
We are not maintaining vendor folder anywhere in this repo, so it does not make much sense to have go mod vendor done and then use -mod=vendor. Instead for our use case it is always good to rely on default behaviour use this

```
CGO_ENABLED=0 GOOS=$(shell go env GOOS) GOARCH=$(shell go env GOARCH) \
		go build -a \
		-ldflags '-X main.vendorVersion='"${DRIVER_NAME}-${GIT_COMMIT_SHA}"' -extldflags "-static"' \
		-o ${GOPATH}/bin/${EXE_DRIVER_NAME} \
		./cmd/
```

But it is not a blocker, we can probably take this up in next PR

Reply:
ok, we can see what best practice we need to follow, but this is also fine this may update the vendor folder every time

Reply:
During testing, appending `go mod vendor` was the only way to get this to work. Hence comment above block `"Repository does not contain vendor/modules.txt file so re-build with go mod vendor"`